### ISSUE TITLE
Clarification: Handeling of edge actions and node actions with BLOCKING_TYPE != HARD

### ISSUE THREAD

**Issue by mbeutelspacher** [VDA5050/VDA5050#344]
Regarding the handling of lifetimes of edge and node actions with ``BLOCKING_TYPE != HARD``, I could find the following information in the document:

> Actions that are triggered on nodes can run as long as they need to run and should be self-terminating (e.g., an audio signal that lasts for five seconds or a pick action, that is finished after picking up a load) or formulated pairwise (e.g., "activateWarningLights" and "deactivateWarningLights"), although there may be exceptions.

> The traversal of a node also marks the leaving of the edge leading up to the node.
The edge shall then be removed from the `edgeStates` and the actions that were active on the edge shall be finished.

My question would be now, what  should the robot do if it reaches the end of the edge and the **edge** action is not finished yet.
Should it
1) block until the edge action is finished and then report the outcome of the edge action, i.e. don't start any node action until the edge action is finished and also don't remove the edgeState until then
2) cancel the edge action when reaching the node and reporting the edge action as ``FAILED``
3) cancel the edge action when reaching the node and reporting the edge action as ``FINISHED``

(1) sounds reasonable but would make it hard to have actions like "Blink red while on edge XYZ", though that could be handled with node actions on the start and end node.
(2) will result in a lot of failed actions, because the robot traversed the edge _too fast_.
(3) would be the solution for the "Blink red while on edge XYZ" edge action, but definitely feels wrong.

Regarding **node** actions with ``BLOCKING_TYPE=NONE`` I would assume, they are allowed to continue until they conflict with a ``BLOCKING_TYPE=HARD`` in which case we would need to wait for the action to finish before starting the ``BLOCKING_TYPE=HARD`` action. For node actions with ``BLOCKING_TYPE=SOFT`` I would assume the same is true, but we would also need to block until completion before the robot is allowed to continue driving (i.e. edge transition).

Clarification would be highly appreciated!

**Comment by ConstantinUp** [comment]
For version 3.0 we did a rework of the blockingType definition. Please check out here: [6.2.2 Action blocking types and sequence](https://github.com/VDA5050/VDA5050/blob/dev/V3.0.0/VDA5050_EN.md#622-action-blocking-types-and-sequence).

In addition in the core team discussion today, it was decided that weather an action needs to be finished for the order to continue (in my opinion also an edge action as you described) is up to the implementation between fleet control an the mobile robot. This means your proposals 1. and 3. are valid. Which of those two is the correct behavior is up to the implementation of this actionType.

Please reopen/add new issue if with version 3.0 something is unclear.
